libraries/audUtils/bidderUtils.js

```diff
@@ -0,0 +1,129 @@
+import {
+  deepSetValue,
+  generateUUID,
+  logError
+} from '../../src/utils.js';
+
+// Function to get Request
+export const getBannerRequest = (bidRequests, bidderRequest, ENDPOINT) => {
+  let request = [];
+  // Loop for each bid request
+  bidRequests.forEach(bidReq => {
+    let guid = generateUUID();
+    const req = {
+      id: guid,
+      imp: [getImpDetails(bidReq)],
+      placementId: bidReq.params.placement_id,
+      site: getSiteDetails(bidderRequest),
+      user: getUserDetails(bidReq)
+    };
+    // Fetch GPP Consent from bidderRequest
+    if (bidderRequest && bidderRequest.gppConsent && bidderRequest.gppConsent.gppString) {
+      deepSetValue(req, 'regs.gpp', bidderRequest.gppConsent.gppString);
+      deepSetValue(req, 'regs.gpp_sid', bidderRequest.gppConsent.applicableSections);
+    } else if (bidderRequest && bidderRequest.ortb2 && bidderRequest.ortb2.regs && bidderRequest.ortb2.regs.gpp) {
+      deepSetValue(req, 'regs.gpp', bidderRequest.ortb2.regs.gpp);
+      deepSetValue(req, 'regs.gpp_sid', bidderRequest.ortb2.regs.gpp_sid);
+    }
+    // Fetch coppa compliance from bidderRequest
+    if (bidderRequest && bidderRequest.ortb2 && bidderRequest.ortb2.regs && bidderRequest.ortb2.regs.coppa) {
+      deepSetValue(req, 'regs.coppa', 1);
+    }
+    // Fetch uspConsent from bidderRequest
+    if (bidderRequest?.uspConsent) {
+      deepSetValue(req, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+    }
+    request.push(req);
+  });
+  // Return the array of request
+  return {
+    method: 'POST',
+    url: ENDPOINT,
+    data: JSON.stringify(request),
+    options: {
+      contentType: 'application/json',
+    }
+  };
+}
+// Function to get Response
+export const getBannerResponse = (bidResponse, mediaType) => {
+  let responseArray = [];
+  if (bidResponse) {
+    try {
+      let bidResp = bidResponse.body && bidResponse.body.seatbid && bidResponse.body.seatbid[0] ? bidResponse.body.seatbid[0].bid : [];
+      if (bidResp) {
+        bidResp.forEach(bidReq => {
+          let response = {};
+          response.requestId = bidReq.impid ? bidReq.impid : undefined;
+          response.cpm = bidReq.price ? bidReq.price : 0.0;
+          response.width = bidReq.w ? bidReq.w : 0;
+          response.height = bidReq.h ? bidReq.h : 0;
+          response.ad = bidReq.adm ? bidReq.adm : '';
```

Comment:
If your bid response doesn't have the `adm` field there is no ad to display. I'm not sure setting it to an empty string would be a good result for publishers.

For most of these properties in the response fields we would expect your response to have them 100% of the time, so you should add validation to this before adding it and returning in the responseArray